MapboxNavigation/Route.swift

```diff
@@ -41,4 +41,62 @@ extension Route {
             return LineString(trimmedPrecedingCoordinates + followingPolyline.trimmed(from: followingPolyline.coordinates[0], distance: distance)!.coordinates.suffix(from: 1))
         }
     }
+
+    var tollIntersections: [Intersection]? {
+        let allSteps = legs.flatMap { return $0.steps }
+
+        let allIntersections = allSteps.compactMap { return $0.intersections }.reduce([], +)
+        let intersectionsWithTolls = allIntersections.filter { return $0.tollCollection != nil }
+
+        return intersectionsWithTolls
+    }
+
+    // returns the list of line segments along the route that fall within given bounding box. Returns nil if there are none. Line segments are defined by the route shape coordinates that lay within the bounding box
+    func shapes(within: Turf.BoundingBox) -> [LineString]? {
+        guard let coordinates = shape?.coordinates else { return nil }
+        var lines = [[CLLocationCoordinate2D]]()
+        var currentLine: [CLLocationCoordinate2D]?
+        for coordinate in coordinates {
```

Comment:
Another approach would be to pass `coordinates` and `coordinates.suffix(from: 1)` into `zip(_:_:)`, then convert the resulting pairs to LineStrings, then map them to `BoundingBox.contains(_:ignoreBoundary:)` or `intersection(_:_:)`. I’m not sure if this approach would be more performant, but it could be more elegant and reusable.

Reply:
Still reviewing this now.